### PR TITLE
Allow users to sym-link a locally installed Capital Framework repo

### PR DESCRIPTION
**Pull request by jimmynotjim**
When working on multiple Capital Framework components, it's difficult to sym-link every component. We already have a NPM link script in the canary branch, this does the other half of NPM linking.

## Additions

- NPM linking and unlinking scripts

## Testing

- To symlink your local CF repo, you'll need to have a duplicate CF repo (I call my cf-site and only have the gh-pages branch available).
- In your main CF repo, run `npm run cf-link` to create the locally available packages
- In your gh-pages specific repo, run `npm run cf-link` to create the second half of the links. 
- In your gh-pages specific repo, run `npm run build && npm start` (~~this may fail due to some variable changes that are outstanding~~ __merged and rebased__). You should now be able to make changes to the usage docs or the source code in the main repo, and see them in the secondary repo. Note that you'll probably need to kick off the appropriate build task in the secondary repo, as browserSync doesn't watch the `node_modules` directory for changes. For usage files it's `gulp copy:usage` for styles it's `gulp styles`.
- To reset your secondary repo, run `npm run cf-unlink`. This will remove the sym-link and reinstall the latest version of each component on NPM.

## Review

- @Scotchester 
- @contolini 
- @anselmbradford 

## Notes

Should we move the Gulp scripts to this `/scripts` directory so it matches the main canary branch?

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: scripts/npm/link/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+var path = require( 'path' );
+var fs = require( 'fs' );
+var componentsDir = path.join( __dirname, '..', '..', '..', 'components' );
```
Comment (Scotchester) [inline scripts/npm/link/index.js]: I'm slightly worried about this breaking if we change the directory structure of the site, but it's not a huge deal.

**Comment by Scotchester** [comment]
⚠️ **Hold** while Jimmy works on automating that `componentsDir` thing I commented on.

File: scripts/npm/link/index.js (hunk 1 above)
Comment (jimmynotjim) [inline scripts/npm/link/index.js]: Mmm, good point. I really want to spend some time adding some unit testing soon. That would catch an issue like this.

**Comment by jimmynotjim** [comment]
@Scotchester realized I already do programmatically pull the list of components on this side, just from the `/components` directory rather than the npm manifest.

**Comment by Scotchester** [comment]
Ah, right. I was conflating this with the other similar PR.

**Comment by Scotchester** [comment]
:+1: But do keep the unit testing you mentioned in mind for the future.

**Pull request merged**